File: app/src/main/java/com/example/tutorial/CalculatorActivity.java
package com.example.tutorial;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.content.res.Configuration;
import android.graphics.Color;
import android.os.Bundle;
import android.os.Handler;
import android.text.TextUtils;
import android.text.method.ScrollingMovementMethod;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.TextView;
import android.widget.Toast;

public class CalculatorActivity extends AppCompatActivity {



    private EditText editTextFirstNum, editTextSecondNum;
    private RadioButton plus, minus, divide, multiply;
    private Button calculate_result;
    private TextView resultTextView;
    private Handler handler;
    private char operation;
    private short counter;
    private final float TEXT_SIZE = 0f;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_calculator);



        editTextFirstNum = findViewById(R.id.editTextFirstNum);
        editTextSecondNum = findViewById(R.id.editTextSecondNum);

        plus = findViewById(R.id.radioButtonPlus);
        plus.setOnClickListener(radioButtonClickListener);

        minus = findViewById(R.id.radioButtonMinus);
        minus.setOnClickListener(radioButtonClickListener);


        divide = findViewById(R.id.radioButtonDivide);
        divide.setOnClickListener(radioButtonClickListener);


        multiply = findViewById(R.id.radioButtonMultiply);
        multiply.setOnClickListener(radioButtonClickListener);


        calculate_result = findViewById(R.id.buttonCalculateResult);


        resultTextView = findViewById(R.id.resultTextView);

        handler = new Handler();

        switch (getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK) {
            case Configuration.UI_MODE_NIGHT_YES:
                resultTextView.setTextColor(ContextCompat.getColor(getApplicationContext(), R.color.white));
                break;
            case Configuration.UI_MODE_NIGHT_NO:
                break;
        }




        calculate_result.setOnClickListener(v -> {
            Toast toast;
            if (TextUtils.isEmpty(editTextFirstNum.getText().toString()) || TextUtils.isEmpty(editTextSecondNum.getText().toString())){
                toast = Toast.makeText(getApplicationContext(), "One of numbers is null.", Toast.LENGTH_SHORT);
                toast.show();
            }
            else if (Float.parseFloat(editTextSecondNum.getText().toString()) == 0f && operation == '/'){
                toast = Toast.makeText(getApplicationContext(), "Can\'t divide by zero", Toast.LENGTH_SHORT);
                toast.show();
            }
            else if (counter == 0){
                toast = Toast.makeText(getApplicationContext(), "Didn\'t chose an operation", Toast.LENGTH_SHORT);
                toast.show();
            }
            else {
                counter = 1;
                new Thread(new Runnable() {
                    boolean isPrime = false;
                    @Override
                    public void run() {
                        float first_num, second_num, res;


                        first_num = Float.parseFloat(editTextFirstNum.getText().toString());
                        second_num = Float.parseFloat(editTextSecondNum.getText().toString());

                        res = getCalculatedResult(first_num, second_num, operation);
                        if (res % 1 == 0){
                            isPrime = true;
                        }


                        handler.post(new Runnable() {
                            @Override
                            public void run() {
                                if (isPrime){
                                    long resLong = (long) res;
                                    Log.d("####", String.valueOf(resLong));
                                    resultTextView.setText(resLong + "");
                                    isPrime = false;
                                }
                                else {
                                    Log.d("####", String.valueOf(res));
                                    resultTextView.setText(res + "");
                                }
                            }
                        });
                    }
                }).run();
            }

        });


    }

    View.OnClickListener radioButtonClickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            counter++;
            RadioButton rb = (RadioButton) v;
            switch (rb.getId()) {
                case R.id.radioButtonPlus: operation = '+';
                    break;
                case R.id.radioButtonMinus: operation = '-';
                    break;
                case R.id.radioButtonDivide: operation = '/';
                    break;
                case R.id.radioButtonMultiply: operation = '*';
                    break;
                default:
                    break;
            }
        }
    };


    private float getCalculatedResult(float a, float b, char operation) {
        float result = 0f;

        switch (operation){
            case '+': result = a + b;
                break;
            case '-': result = a - b;
                break;
            case '/': result = a / b;
                break;
            case '*': result = a * b;
                break;
        }
        return result;
    }
}

File: app/src/main/java/com/cruelgeroi/apphub/activities/CalculatorActivity.kt
package com.cruelgeroi.apphub.activities

import android.content.res.Configuration
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.core.content.ContextCompat
import android.text.TextUtils
import android.util.Log
import android.widget.*
import com.cruelgeroi.apphub.R

class CalculatorActivity : AppCompatActivity() {

    private var mRadioGroup: RadioGroup? = null
    private var editTextFirstNum: EditText? = null
    private var editTextSecondNum: EditText? = null
    private var calculateResult: Button? = null
    private var resultTextView: TextView? = null

    private var operation = '0'




    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_calculator)

        var counter: Short = 0


        mRadioGroup = findViewById(R.id.operationsRadioGroup)
        mRadioGroup?.setOnCheckedChangeListener { _, optionId ->
            run {
                counter++
                when (optionId) {
                    R.id.radioButtonPlus -> operation = '+'
                    R.id.radioButtonMinus -> operation = '-'
                    R.id.radioButtonDivide -> operation = '/'
                    R.id.radioButtonMultiply -> operation = '*'
                }
            }
        }

        editTextFirstNum = findViewById(R.id.editTextFirstNum)
        editTextSecondNum = findViewById(R.id.editTextSecondNum)


        calculateResult = findViewById(R.id.buttonCalculateResult)
        resultTextView = findViewById(R.id.resultTextView)


        when (resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK) {
            Configuration.UI_MODE_NIGHT_YES -> resultTextView?.setTextColor(ContextCompat.getColor(applicationContext, R.color.white))
            Configuration.UI_MODE_NIGHT_NO -> {}
        }

        calculateResult?.setOnClickListener{
            if (TextUtils.isEmpty(editTextFirstNum?.text.toString()) || TextUtils.isEmpty(editTextSecondNum?.text.toString())) {
                Toast.makeText(applicationContext, "One of numbers is null.", Toast.LENGTH_SHORT).show()
            }
            else if (editTextSecondNum?.text.toString().toFloat() == 0f && operation == '/') {
                Toast.makeText(applicationContext, "Can\'t divide by zero", Toast.LENGTH_SHORT).show()
            }
            else if (counter.toInt() == 0) {
                Toast.makeText(applicationContext, "Didn\'t chose an operation", Toast.LENGTH_SHORT).show()
            }
            else {
                Thread(object : Runnable {
                    var isPrime = false
                    override fun run() {
                        val firstNum: Float = editTextFirstNum?.text.toString().toFloat()
                        val secondNum: Float = editTextSecondNum?.text.toString().toFloat()
                        val res: Float = getCalculatedResult(firstNum, secondNum, operation)
                        if (res % 1 == 0f) {
                            isPrime = true
                        }
                        Handler(Looper.getMainLooper()).post {
                            if (isPrime) {
                                val resLong = res.toLong()
                                Log.d("####", resLong.toString())
                                resultTextView?.text = resLong.toString() + ""
                                isPrime = false
                            } else {
                                Log.d("####", res.toString())
                                resultTextView?.text = res.toString() + ""
                            }
                        }
                    }
                }).run()
            }
        }
    }



    private fun getCalculatedResult(a: Float, b: Float, operation: Char): Float {
        var result = 0f
        when (operation) {
            '+' -> result = a + b
            '-' -> result = a - b
            '/' -> result = a / b
            '*' -> result = a * b
        }
        return result
    }
}